### PR TITLE
Redirect into closest parent

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Improves the behavior of the new category redirection function. Now it will redirect into the closest active parent of the current category, showing the customer much more relevant data.
| Type?             | improvement
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Make a category `Home/Clothes/Men/Shirts/Small`, disable `Shirts` and `Small`, visit `Small`, see that you get redirected into `Men`. Watch out, redirection responses are cached in browsers, try different category if it behaves weird.
| UI Tests          | https://github.com/florine2623/testing_pr/actions/runs/10684056246
| Fixed issue or discussion?     | 
| Related PRs       | 
| Sponsor company   | TRENDO s.r.o.

### CODE REVIEW

File: controllers/front/listing/CategoryController.php [hunk 1]
```diff
@@ -348,4 +348,31 @@ public function getListingLabel()
...
         );
     }
+
+    /**
+     * Returns a category that we will redirect into, in case we need 301/302 redirect.
+     * We will try to get the closest parent of the current category.
+     *
+     * @return int category ID
+     */
+    private function getCategoryToRedirectTo()
```
Comment (FabienPapet) [inline controllers/front/listing/CategoryController.php]: :int

**Review by kpodemski — DISMISSED** [review]

**Comment by ps-jarvis** [comment]
This pull request seems to contain new translation strings. I have summarized them below to ease up review:

- `Admin.Catalog.Help`
    - [ ] `By default, the closest active parent category will be used if no category is selected.`

(Note: this is an automated message, but answering it will reach a real human)

**Comment by Hlavtox** [comment]
@kpodemski @FabienPapet Tiny improvement, it's ready. :-)

**Review by kpodemski — APPROVED** [review]

**Review by ShaiMagal — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @Hlavtox ,

Tested the standard case, it works as expected :

https://github.com/user-attachments/assets/85a8497c-5cd8-4ca8-97e8-fb4152ec75ee

Tested the case when 1 or more subcategories are disabled.
Tested the case when the first category is disabled.

LGTM
It is QA ✅

**Comment by FabienPapet** [comment]
TY @Hlavtox

**Pull request merged**